substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/phases/IntrinsifyMethodHandlesInvocationPlugin.java

```diff
@@ -516,12 +524,26 @@ private void processInvokeWithMethodHandle(GraphBuilderContext b, Replacements r
 
             JavaKind returnResultKind = b.getInvokeReturnType().getJavaKind().getStackKind();
             ValueNode transplantedSingleFunctionality = null;
+            NewInstanceNode transplantedNewInstance = null;
             if (singleFunctionality instanceof Invoke) {
```

Comment:
What happens if there is a `singleNewInstance` but `singleFunctionality` is not an `Invoke`?

Reply:
The processInvokeWithMethodHandle should always have an Invoke but I can add an assert.

Reply:
Added a check for this situation.